commons/src/main/java/io/aiven/kafka/tieredstorage/commons/ChunkManager.java

```diff
@@ -18,18 +18,52 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.List;
 
 import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
 
 import io.aiven.kafka.tieredstorage.commons.manifest.SegmentManifest;
+import io.aiven.kafka.tieredstorage.commons.security.AesEncryptionProvider;
+import io.aiven.kafka.tieredstorage.commons.storage.ObjectStorageFactory;
+import io.aiven.kafka.tieredstorage.commons.transform.BaseDetransformChunkEnumeration;
+import io.aiven.kafka.tieredstorage.commons.transform.DecompressionChunkEnumeration;
+import io.aiven.kafka.tieredstorage.commons.transform.DecryptionChunkEnumeration;
+import io.aiven.kafka.tieredstorage.commons.transform.DetransformChunkEnumeration;
+import io.aiven.kafka.tieredstorage.commons.transform.DetransformFinisher;
+
+public class ChunkManager {
+    private final ObjectStorageFactory objectStorageFactory;
+    private final ObjectKey objectKey;
+    private final AesEncryptionProvider aesEncryptionProvider;
+
+    public ChunkManager(final ObjectStorageFactory objectStorageFactory, final ObjectKey objectKey,
+            final AesEncryptionProvider aesEncryptionProvider) {
+        this.objectStorageFactory = objectStorageFactory;
+        this.objectKey = objectKey;
+        this.aesEncryptionProvider = aesEncryptionProvider;
+    }
 
-public interface ChunkManager {
     /**
      * Gets a chunk of a segment.
      *
      * @return an {@link InputStream} of the chunk, plain text (i.e. decrypted and decompressed).
      */
-    InputStream getChunk(RemoteLogSegmentMetadata remoteLogSegmentMetadata,
-                         SegmentManifest manifest,
-                         int chunkId) throws IOException;
+    public InputStream getChunk(final RemoteLogSegmentMetadata remoteLogSegmentMetadata, final SegmentManifest manifest,
```

Comment:
Could we add some tests for getting chunk, e.g. pipeline ordering (decryption and then decompression)